OpenRA.Mods.Common/Traits/BotModules/BaseBuilderBotModule.cs

```diff
@@ -245,6 +273,48 @@ void IBotTick.BotTick(IBot bot)
 				}
 			}
 
+			if (--checkBestResourceLocationTicks <= 0 && resourceLayer != null)
+			{
+				checkBestResourceLocationTicks = Info.CheckBestResourceLocationInterval;
+
+				// Get rid of request that add too many refinery to a map indice
+				if (ResourceMapModule != null)
+				{
+					foreach (var mcv in RequestedRefineries.Keys.ToList())
+					{
+						if (ResourceMapModule.FindClosestIndiceFromCPos(
+							RequestedRefineries[mcv].ResourceLoc).PlayerRefineryCount >= Info.MaxRefineryPerIndice)
+							RequestedRefineries.Remove(mcv);
+					}
+				}
+
+				Actor bestconyard = null;
+				var best = int.MinValue;
+
+				foreach (var conyard in ConstructionYardBuildings.Actors)
+				{
+					if (conyard.IsDead || !world.Map.FindTilesInAnnulus(conyard.Location, Info.MinBaseRadius, Info.MaxBaseRadius)
```

Comment:
```suggestion
					if (conyard.IsDead)
						continue;

					if (!world.Map.FindTilesInAnnulus(conyard.Location, Info.MinBaseRadius, Info.MaxBaseRadius)
						.Any(c => ResourceMapModule != null
						? ResourceMapModule.Info.ValuableResourceTypes.Contains(resourceLayer.GetResource(c).Type)
						: resourceLayer.GetResource(c).Type != null))
						continue;
```

for better readability?